openedx/core/djangoapps/schedules/management/commands/tests/test_send_upgrade_reminder.py

```diff
@@ -1,427 +1,81 @@
 
 
-SITE_QUERY = 1
-SCHEDULES_QUERY = 1
-COURSE_MODES_QUERY = 1
-GLOBAL_DEADLINE_SWITCH_QUERY = 1
-COMMERCE_CONFIG_QUERY = 1
-NUM_QUERIES_NO_ORG_LIST = 1
-
-NUM_QUERIES_NO_MATCHING_SCHEDULES = SITE_QUERY + SCHEDULES_QUERY
-
-NUM_QUERIES_WITH_MATCHES = (
-    NUM_QUERIES_NO_MATCHING_SCHEDULES +
-    COURSE_MODES_QUERY
-)
-
-NUM_QUERIES_FIRST_MATCH = (
-    NUM_QUERIES_WITH_MATCHES
-    + GLOBAL_DEADLINE_SWITCH_QUERY
-    + COMMERCE_CONFIG_QUERY
-)
-
 LOG = logging.getLogger(__name__)
 
 
 @ddt.ddt
 @skip_unless_lms
 @skipUnless('openedx.core.djangoapps.schedules.apps.SchedulesConfig' in settings.INSTALLED_APPS,
             "Can't test schedules if the app isn't installed")
-@freeze_time('2017-08-01 00:00:00', tz_offset=0, tick=True)
-class TestUpgradeReminder(SharedModuleStoreTestCase):
-
-    ENABLED_CACHES = ['default']
-
-    @classmethod
-    def setUpClass(cls):
-        super(TestUpgradeReminder, cls).setUpClass()
-
-        cls.course = CourseFactory.create(
-            org='edX',
-            number='test',
-            display_name='Test Course',
-            self_paced=True,
-            start=datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=30),
-        )
-        cls.course_overview = CourseOverview.get_from_id(cls.course.id)
-
-    def setUp(self):
-        super(TestUpgradeReminder, self).setUp()
-
-        CourseModeFactory(
-            course_id=self.course.id,
-            mode_slug=CourseMode.VERIFIED,
-            expiration_datetime=datetime.datetime.now(pytz.UTC) + datetime.timedelta(days=30),
-        )
-
-        ScheduleFactory.create(upgrade_deadline=datetime.datetime(2017, 8, 1, 15, 44, 30, tzinfo=pytz.UTC))
-        ScheduleFactory.create(upgrade_deadline=datetime.datetime(2017, 8, 1, 17, 34, 30, tzinfo=pytz.UTC))
-        ScheduleFactory.create(upgrade_deadline=datetime.datetime(2017, 8, 2, 15, 34, 30, tzinfo=pytz.UTC))
-
-        site = SiteFactory.create()
-        self.site_config = SiteConfigurationFactory.create(site=site)
-        ScheduleConfigFactory.create(site=self.site_config.site)
+class TestUpgradeReminder(ScheduleBaseEmailTestBase):
+    __test__ = True
 
-        DynamicUpgradeDeadlineConfiguration.objects.create(enabled=True)
+    tested_task = tasks.ScheduleUpgradeReminder
+    deliver_task = tasks._upgrade_reminder_schedule_send
+    tested_command = reminder.Command
+    deliver_config = 'deliver_upgrade_reminder'
+    enqueue_config = 'enqueue_upgrade_reminder'
+    expected_offsets = (2,)
 
-    @patch.object(reminder.Command, 'async_send_task')
-    def test_handle(self, mock_send):
-        test_day = datetime.datetime(2017, 8, 1, tzinfo=pytz.UTC)
-        reminder.Command().handle(date='2017-08-01', site_domain_name=self.site_config.site.domain)
-        mock_send.enqueue.assert_called_with(
-            self.site_config.site,
-            test_day,
-            2,
-            None
-        )
+    has_course_queries = True
```

Comment:
Now that my PR is merged, I think we will also need a `has_org_queries`

Reply:
ok.  thanks for the warning.